cmd/cstor-pool-mgmt/controller/new-pool-controller/run_pool_controller.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package poolcontroller
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/cmd/cstor-pool-mgmt/controller/common"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Run will set up the event handlers for types we are interested in, as well
+// as syncing informer caches and starting workers. It will block until stopCh
+// is closed, at which point it will shutdown the workqueue and wait for
+// workers to finish processing their current work items.
+func (c *CStorPoolController) Run(threadiness int, stopCh <-chan struct{}) error {
+	defer runtime.HandleCrash()
+	defer c.workqueue.ShutDown()
+
+	// Start the informer factories to begin populating the informer caches
+	glog.Info("Starting CStorPool controller")
+
+	// Wait for the k8s caches to be synced before starting workers
+	glog.Info("Waiting for informer caches to sync")
+	if ok := cache.WaitForCacheSync(stopCh, c.cStorPoolSynced); !ok {
+		return fmt.Errorf("failed to wait for caches to sync")
+	}
+	glog.Info("Starting CStorPool workers")
+	// Launch worker to process CStorPool resources
+	for i := 0; i < threadiness; i++ {
+		go wait.Until(c.runWorker, common.ResourceWorkerInterval, stopCh)
+	}
+
+	glog.Info("Started CStorPool workers")
+	<-stopCh
```

Comment:
old controller code related to 'waiting for cstor-pool container to come up' are removed in this..
this need to be handled in the handler code..
all the import checks can fail, and by the time, handler reaches create code and if cstor-pool comes up by that time, it ends up creating pool that can lead to data loss..
so, lets make sure 'zpool status' works as first step in handler code..

Reply:
this also removed the 'liveness check for cstor-pool' kind of thing which used to make cstor-pool-mgmt as well panic faster..
but, now, current code depends on sync handler to import the pool if cstor-pool crashes in between